### PR TITLE
Add initial_pan option to set the starting pan position

### PR DESCRIPTION
**Pull request by DarkDefender**
My initial take on: #176

### CODE REVIEW

File: doc/imv.5.txt [hunk 1]
```diff
@@ -42,6 +42,11 @@ The *[options]* section accepts the following settings:
 *height* = <height>::
 	Initial height of the imv window. Defaults to 720.
 
+*initial_pan* = <pan_factor_x> <pan_factor_y>::
+	Initial pan/focus position factor of the opened images. A value of 50
+	reprecents the middle point of the image (50%).
```
Comment (eXeC64) [inline doc/imv.5.txt]: Typo here, you want `represents`.

File: src/imv.c [hunk 2]
```diff
@@ -479,6 +483,7 @@ struct imv *imv_create(void)
   imv->need_redraw = true;
   imv->need_rescale = true;
   imv->scaling_mode = SCALING_FULL;
+  imv->custom_start_pan = false;
```
Comment (eXeC64) [inline src/imv.c]: `imv` is explicitly zero-initialised, so `0` values don't need to be set. Please take this line out.

File: src/viewport.c [hunk 3]
```diff
@@ -30,6 +31,7 @@ struct imv_viewport *imv_viewport_create(int window_width, int window_height,
   view->buffer.height = buffer_height;
   view->scale = 1;
   view->x = view->y = view->redraw = 0;
+  view->pan_factor_x = view->pan_factor_y = 0.5f;
```
Comment (eXeC64) [inline src/viewport.c]: Just `0.5` for this, no `f` since it's a double not a float being set.

**Comment by DarkDefender** [comment]
The behavior for the corner is correct in this case.

It works like this currently:
```
___________________________
 0 0   | 50 0   | 100 0   |
---------------------------
 0 50  | 50 50  | 100 50  |
---------------------------
 0 100 | 50 100 | 100 100 |
---------------------------
```

So if you want it to be centered on zoom out, you want to have `initial_pan` set to `50 0`
Is this ok with you?

**Comment by eXeC64** [comment]
By centred I meant both vertically and horizontally. My expectation here would be that `initial_pan` would only take effect if the image is larger than the window in either dimension. The use case for this is starting in a particular area of the image if it's not all visible to begin with, rather than pinning the image to the edge regardless of context.

I'm just not sure why you'd want the image to appear in the top corner by default if it's smaller than the window.

**Comment by DarkDefender** [comment]
Ah, I see. My bad I misunderstood you. I'll look into it.

**Comment by DarkDefender** [comment]
It should now center the image if it is smaller than the window.

**Comment by eXeC64** [comment]
Nice, thanks! I've applied the patch to master.

**Pull request closed without merging**